packages/nimble-components/src/wafer-map/modules/data-manager.ts

```diff
@@ -56,36 +49,34 @@ export class DataManager {
 
     private readonly computations: Computations;
     private readonly prerendering: Prerendering;
+    private readonly dataMap: Map<string, WaferMapDie>;
 
-    public constructor(
-        dies: Readonly<Readonly<WaferMapDie>[]>,
-        axisLocation: Readonly<WaferMapQuadrant>,
-        canvasDimensions: Readonly<Dimensions>,
-        colorScale: Readonly<WaferMapColorScale>,
-        highlightedValues: Readonly<string[]>,
-        colorScaleMode: Readonly<WaferMapColorScaleMode>,
-        dieLabelsHidden: Readonly<boolean>,
-        dieLabelsSuffix: Readonly<string>,
-        maxCharacters: Readonly<number>
-    ) {
-        this.computations = new Computations(
-            dies,
-            axisLocation,
-            canvasDimensions
-        );
+    public constructor(wafermap: WaferMap) {
+        this.computations = new Computations(wafermap.dies, wafermap.quadrant, {
+            width: wafermap.canvasSideLength,
+            height: wafermap.canvasSideLength
+        });
 
         this.prerendering = new Prerendering(
-            dies,
-            colorScale,
-            highlightedValues,
+            wafermap.dies,
+            wafermap.colorScale,
+            wafermap.highlightedValues,
             this.computations.horizontalScale,
             this.computations.verticalScale,
-            colorScaleMode,
-            dieLabelsHidden,
-            dieLabelsSuffix,
-            maxCharacters,
+            wafermap.colorScaleMode,
+            wafermap.dieLabelsHidden,
+            wafermap.dieLabelsSuffix,
+            wafermap.maxCharacters,
             this.computations.dieDimensions,
             this.computations.margin
         );
+
+        this.dataMap = new Map(
+            wafermap.dies.map(die => [`${die.x}_${die.y}`, die])
+        );
+    }
+
+    public getWaferMapDie(x: number, y: number): WaferMapDie | undefined {
+        return this.dataMap.get(`${x}_${y}`);
     }
```

Comment:
This will be eventually used by the HoverHandler, and the intent was to leave it there as it is, instead of removing and re-adding it later in a different PR. 
In the interest of time I will remove it though from the current PR.